cyclops-ui/src/components/k8s-resources/ClusterRole.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { useCallback, useEffect, useState } from "react";
+import { Col, Divider, Row, Alert, Table, Tag } from "antd";
+import axios from "axios";
+import { mapResponseError } from "../../utils/api/errors";
+import { isStreamingEnabled } from "../../utils/api/common";
+
+interface Props {
+  name: string;
+}
+
+interface Rule {
+  verbs: string[];
+  apiGroups: string[];
+  resources: string[];
+}
+
+interface ClusterRoleData {
+  rules: Rule[];
+}
+
+const ClusterRole = ({ name }: Props) => {
+  const [clusterRole, setClusterRole] = useState<ClusterRoleData>({
+    rules: [],
+  });
+
+  const [error, setError] = useState({
+    message: "",
+    description: "",
+  });
+
+  const fetchClusterRole = useCallback(() => {
+    axios
+      .get(`/api/resources`, {
+        params: {
+          group: `rbac.authorization.k8s.io`,
+          version: `v1`,
+          kind: `ClusterRole`,
+          name: name,
+        },
+      })
+      .then((res) => {
+        setClusterRole({
+          rules: res.data.rules || [],
+        });
+      })
+      .catch((error) => {
+        setError(mapResponseError(error));
+      });
+  }, [name]);
+
+  useEffect(() => {
+    fetchClusterRole();
+
+    if (isStreamingEnabled()) {
+      return;
+    }
+
+    const interval = setInterval(() => fetchClusterRole(), 15000);
+    return () => {
+      clearInterval(interval);
+    };
+  }, [fetchClusterRole]);
+
+  const columns = [
+    {
+      title: "API Groups",
+      dataIndex: "apiGroups",
+      key: "apiGroups",
+      render: (apiGroups: string[]) => (
+        <>
+          {apiGroups.map((group) => (
+            <Tag key={group} color="blue">
+              {group || "*"}
+            </Tag>
+          ))}
```

Comment:
Not all rules will have `apiGroups`. With cluster roles, you can have a rule that has a `nonResourceURLs` field, and those rules will not have the `apiGroups` property.

You can create a Module from the Prometheus bitnami helm chart and you will get one such cluster role which  has a single rule without `apiGroups`.

Because of that, the `apiGroups` here will be null and calling `.map` will break the UI

Reply:
Yes right. I've fixed this case.

![image](https://github.com/user-attachments/assets/0096d181-a2e8-48db-a555-b0f75de2b128)

Should we display something for this rule ?

```json
{
	"2": {
		"verbs": [
			"get"
		],
		"nonResourceURLs": [
			"/metrics"
		]
	}
}
```

Reply:
It might be a bit confusing if you see a rule without any resources and just the verbs.
Can we conditionally add a column to this table? I was thinking of adding a nonResourceURLs column if there is at least one rule with the property. If not, I would leave those three columns. In that case, I would put the verbs as the first column, followed by APIGroups and resources and, if needed, nonREsourceURLs.

Reply:
@nikonhub, can you check my comment above and add a column for nonResourceURLs if possible